main.cpp

```diff
@@ -51,22 +51,322 @@ If you need inspiration for what to write, take a look at previously approved st
+    string castBanish(string thingToBeBanished);
+    int castIllusions(int numIllusions = 1);
+};
+
+Preistess::Preistess() :
+favoriteFood("Salad"),
+leastFavoriteThing("Bad Smells"),
+crystals(12),
+illusions(20),
+home("Tree house")
+{
+    cout << "CONSTRUCTING Preistess with " << crystals << " ctrystals"  << endl;
+}
+
+Preistess::~Preistess()
+{
+    cout << "Deconstructing Preistess" << endl;
+}
+
+int Preistess::castGrowSpell(int numPlants)
+{
+    for (int i = 0; i < Preistess::crystals; i += 1)
+    {
+        numPlants += 1;
+    }
+    cout << "Priestess casts grow and " << numPlants << " plants grow immediately" << endl;
+    return numPlants;
+}
 
+string Preistess::castBanish(string thingToBeBanished)
+{
+    if (thingToBeBanished != Preistess::leastFavoriteThing)
+    {
+        cout << "Preistess casts banish and " << thingToBeBanished << " are banished for 2 rounds" << endl;
+        return "praretess casts banish and " + thingToBeBanished + " are banished for 2 rounds";
+    }
+    cout << "Preistess casts banish and " << Preistess::leastFavoriteThing << " are banished for 2 rounds" << endl;
+        return "preistess casts banish and " + Preistess::leastFavoriteThing + " are banished for 2 rounds";
+}
+
+int Preistess::castIllusions(int numIllusions)
+{
+    int index = Preistess::crystals;
+    while (index > 0)
+    {
+        numIllusions *= 2;
+        index --;
+    }
+    cout << "Preistess casts Illusions and " << numIllusions << " illusions appear instantly" << endl;
+    return numIllusions;
+}
 /*
  new UDT 4:
  */
+ struct Thief {
+     Thief();
+     ~Thief();
+     int tools;
+     string guild;
+     bool secretLanguage;
+     double speed = 33.50;
+     class Techniques {
+         Techniques(string main);
+         ~Techniques();
+         string main = "sneak";
+         string second = "climb";
+         string third = "back stab";
+         string fourth = "pick pocket";
+         string fifth = "read lips";
+     };
+     bool cartwheel(int dexterity = 13);
+     bool smoothTalk();
+     float steal(float amount = 50.00f);
+ };
+
+ Thief::Thief() :
+ tools(15),
+ guild("Secret"),
+ secretLanguage(true)
+ {
+     cout << "CONSTRUCTING Thief with " << guild << " guild" << endl;
+ }
+
+ Thief::~Thief()
+ {
+     cout << "DECONSTRUCTING Thief" << endl;
+ }
+
+bool Thief::cartwheel(int dexterity)
+{
+    if (dexterity >= 13  ){
+        cout << "Theif carwheels no problem" << endl;
+        return true;
+    }
+    cout << "Theif fails at a cartwheel and is vulnerable to insults" << endl;
+    return false;
+}
+
+int randomNum() 
+{
+  std::random_device rd;
+  std::mt19937 gen(rd());
+  std::uniform_int_distribution<int> dist(1, 20);
+  int result = dist(gen);
+  return result;
+}
+
+bool Thief::smoothTalk()
+{
+    if (randomNum() > 11){
+        cout << "Theif smooth talks effectively" << endl;
+        return true;
+    }
+    cout << "Theif thinks he's so smooth but is not" << endl;
+    return false;
+}
```

Comment:
curly braces go on their own line with no other code
fix the spelling errors
- [ ] revise